### PR TITLE
[CALCITE-2469] RexSimplify should not optimize 'not(x) is null' to 'x is not null

### PR DESCRIPTION
**Pull request by jiayuanv127**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/RexProgramTest.java [hunk 1]
```diff
@@ -1289,6 +1292,11 @@ private void checkExponentialCnf(int n) {
     checkSimplify(coalesce(iRef, literal1), "COALESCE(?0.i, 1)");
     checkSimplify(coalesce(iRef, plus(iRef, hRef), literal1, hRef),
         "COALESCE(?0.i, +(?0.i, ?0.h), 1)");
+
+    // not(x) is null should not optimized to x is not null
+    checkSimplify(isNull(not(fRef)), "IS NULL(NOT(?0.f))");
```
Comment (vlsi) [inline core/src/test/java/org/apache/calcite/test/RexProgramTest.java]: Please rebase and use `isNull(not(vBool()))`

Comment (vlsi) [reply core/src/test/java/org/apache/calcite/test/RexProgramTest.java]: please add case for isNull(not(vBoolNotNull())) as well

Comment (jiayuanv127) [reply core/src/test/java/org/apache/calcite/test/RexProgramTest.java]: Thanks! I have modified.

File: core/src/main/java/org/apache/calcite/rex/RexSimplify.java [hunk 2]
```diff
@@ -471,6 +471,12 @@ private RexNode simplifyIs2(SqlKind kind, RexNode a) {
     }
     switch (a.getKind()) {
     case NOT:
+      // NOT(x) IS NULL !=> x IS NOT NULL
+      // Similarly NOT(x) IS NOT NULL !=> x IS NULL
+      if (kind == SqlKind.IS_NULL
```
Comment (vlsi) [inline core/src/main/java/org/apache/calcite/rex/RexSimplify.java]: Please remove this and add relevant code to `negateNullSafe`

**Pull request closed without merging**